openflighthpc/carbon-leaderboard
# Leaderboard Initial Implementation & Several Minor Changes

# Overview

This PR mainly implements the default leaderboard. It also made some adjustments to previous styles

# Major Changes

## Front End
- `app/views/report/index.html.erb` is changed.
- `app/assets/stylesheets/leaderboard.css` and `app/javascript/report/index.js` are added to provide logics/styles for the leaderboard page.
- Default modules `turbo-rails` and `controllers` have been disabled since they caused some unknown bugs.

## Back End
- `raw_data` is roughly implemented to respond with the minimum required JSON data for the default leaderboard.

# Minor Changes
- Since the text color `#586674` for the navibar looks better than `#404040` in most cases, it is now defined as the default font color.
- The background style of the navibar has been changed based on personal preferences. Whether it is better than the previous implementation needs to be evaluated by the reviewer.

# Testing
1. Pull this branch and start the server with the command `rails s`.
2. This PR is shipped with a feasible approach to populate instance reports:
   - Uncomment line 2 to line 20 in `app/javascript/report/index.js`.
   - Refresh the webpage `localhost:3000/leaderboards`. A new record should be added to the leaderboard each time the webpage is refreshed until the uncommented lines are disabled again.
3. Once there are enough records. Visit `localhost:3000/leaderboards` to see the result.
